eye2bids/edf2bids.py

```diff
@@ -360,9 +414,26 @@ def edf2bids(
         "TaskName": _extract_TaskName(events),
     }
 
-    with open(output_dir / "_events.json", "w") as outfile:
-        json.dump(events_json, outfile, indent=4)
-    e2b_log.info(f"file generated: {output_dir / '_events.json'}")
+    if substring_events not in filename:
+        with open(output_dir / (filename + "_events.json"), "w") as outfile:
+            json.dump(events_json, outfile, indent=4)
+        e2b_log.info(f"file generated: {output_dir / (filename + '_events.json')}")
+    else:
+        with open(output_dir / (filename + ".json"), "w") as outfile:
+            json.dump(events_json, outfile, indent=4)
+        e2b_log.info(f"file generated: {output_dir / (filename + '.json')}")
+
+    # Samples to eyetrack.tsv
+    eyetrack_tsv = _samples_to_data_frame(samples_asc_file)
+
+    if substring_eyetrack not in filename:
+        with open(output_dir / (filename + "_eyetrack.tsv"), "w") as outfile:
+            eyetrack_tsv.to_csv(outfile, sep="\t", index=False, compression="gzip")
+        e2b_log.info(f"file generated: {output_dir / (filename + '_eyetrack.tsv')}")
+    else:
+        with open(output_dir / (filename + ".tsv"), "w") as outfile:
+            eyetrack_tsv.to_csv(outfile, sep="\t", index=False, compression="gzip")
+        e2b_log.info(f"file generated: {output_dir / (filename + '.tsv')}")
```

Comment:
I don't understand why we may have those different cases of filenaming

Reply:
@mszinte had a dataset where he already had named the files with "_eyetrack" at the end. I don't know if this is super common but I guess it could be that others had the same idea of naming their files like this without knowing BIDS. Then when they want to convert it to BIDS they would get "_eyetrack_eyetrack" as output filename

Reply:
OK fair, I will refactor the code a bit for this rare case